tsdb/db.go

```diff
@@ -1505,34 +1505,38 @@ func (db *DB) CleanTombstones() (err error) {
 	start := time.Now()
 	defer db.metrics.tombCleanTimer.Observe(time.Since(start).Seconds())
 
-	newUIDs := []ulid.ULID{}
-	defer func() {
-		// If any error is caused, we need to delete all the new directory created.
-		if err != nil {
-			for _, uid := range newUIDs {
-				dir := filepath.Join(db.Dir(), uid.String())
-				if err := os.RemoveAll(dir); err != nil {
-					level.Error(db.logger).Log("msg", "failed to delete block after failed `CleanTombstones`", "dir", dir, "err", err)
-				}
-			}
-		}
-	}()
+	// We can momentarily allow overlapping blocks in this particular case.
+	// This will allow the reload function to replace old blocks with the new ones.
+	// Notice that the the final result will be overlapping blocks, but with no tombstones.
+	if !db.opts.AllowOverlappingBlocks {
+		db.opts.AllowOverlappingBlocks = true
+		defer func() { db.opts.AllowOverlappingBlocks = false }()
+	}
```

Comment:
We don't need this. And the final result is not overlapping blocks because the `reload()` will remove the old block.